### DISCUSSION TITLE
[V5] Cannot find module './src/dialects/oracledb'

### DISCUSSION THREAD

**Discussion by mikelogaciuk** [adonisjs/core#2295] [Help] [answered]
I have a with problem Adonis (V5) when connecting it to OracleDb.

I've done everything according by manual, but HealthCheck throws an err (MSSQL works ok):

 `"Cannot find module './src/dialects/oracledb'\nRequire stack:\n-`

![obraz](https://user-images.githubusercontent.com/55132446/109019054-ed2ad780-76b8-11eb-979a-9e77a734bdad.png)

OracleDB works properly in other project with vanilla JS & node/express, so it is not a problem with oracledb configuration in my OS itself.

Full error:

```
"Cannot find module './src/dialects/oracledb'\nRequire stack:\n- 
C:\\xxx\\repos\\freya\\node_modules\\knex-dynamic-connection\\build\\index.js\n- 
C:\\xxx\\repos\\freya\\node_modules\\@adonisjs\\lucid\\build\\src\\Connection\\index.js\n-
C:\\xxx\\repos\\freya\\node_modules\\@adonisjs\\lucid\\build\\src\\Connection\\Manager.js\n- 
C:\\xxx\\repos\\freya\\node_modules\\@adonisjs\\lucid\\build\\src\\Database\\index.js\n- 
C:\\xxx\\repos\\freya\\node_modules\\@adonisjs\\lucid\\build\\providers\\DatabaseProvider.js\n- 
C:\\xxx\\repos\\freya\\node_modules\\@poppinss\\utils\\build\\src\\esmRequire.js\n- 
C:\\xxx\\repos\\freya\\node_modules\\@poppinss\\utils\\build\\index.js\n- 
C:\\xxx\\repos\\freya\\node_modules\\@adonisjs\\require-ts\\build\\src\\Compiler\\index.js\n- 
C:\\xxx\\repos\\freya\\node_modules\\@adonisjs\\require-ts\\build\\index.js\n- 
C:\\xxx\\repos\\freya\\node_modules\\@adonisjs\\assembler\\build\\src\\requireHook\\index.js\n- 
C:\\xxx\\repos\\freya\\node_modules\\@adonisjs\\assembler\\build\\register.js\n- internal/preload"
```

database.ts:
```ts
    oracle: {
      client: 'oracledb',
      connection: {
        host: Env.get('ORACLE_HOST'),
        port: Env.get('ORACLE_PORT'),
        user: Env.get('ORACLE_USER'),
        password: Env.get('ORACLE_PASSWORD', ''),
        database: Env.get('ORACLE_DB_NAME'),
      },
      healthCheck: true,
      debug: false,
    }
```

.env:
```
ORACLE_HOST=XXXX
ORACLE_PORT=1521
ORACLE_USER=XXXX
ORACLE_PASSWORD=XXXX
ORACLE_DB_NAME=dual
```

I've tried to find something in the web, but failed to do so.

**Comment by thetutlage** [comment]
Can you share the version of `@adonisjs/lucid` package?

**Reply by mikelogaciuk** [reply]
@adonisjs/lucid:

```json
"@adonisjs/lucid": "^10.0.0"
```

Whole package.json:

```json
{
  "name": "freya",
  "version": "1.0.0",
  "private": true,
  "scripts": {
    "build": "node ace build --production",
    "start": "node server.js",
    "dev": "node ace serve --watch",
    "lint": "eslint . --ext=.ts",
    "format": "prettier --write ."
  },
  "devDependencies": {
    "@adonisjs/assembler": "^3.0.6",
    "@types/node": "^14.14.31",
    "adonis-preset-ts": "^2.0.0",
    "eslint": "^7.20.0",
    "eslint-config-prettier": "^8.0.0",
    "eslint-plugin-adonis": "^1.2.1",
    "eslint-plugin-prettier": "^3.3.1",
    "pino-pretty": "^4.5.0",
    "prettier": "^2.2.1",
    "typescript": "^4.1.5",
    "youch": "^2.1.1",
    "youch-terminal": "^1.0.1"
  },
  "dependencies": {
    "@adonisjs/core": "^5.0.4-preview-rc-2.1",
    "@adonisjs/lucid": "^10.0.0",
    "@adonisjs/repl": "^1.1.6",
    "@types/mssql": "^6.0.7",
    "@types/oracledb": "^5.1.0",
    "luxon": "^1.26.0",
    "mssql": "^6.3.1",
    "oracledb": "^5.1.0",
    "proxy-addr": "^2.0.6",
    "reflect-metadata": "^0.1.13",
    "source-map-support": "^0.5.19"
  }
}
```

**Comment by mikelogaciuk** [comment, marked as answer]
On the other hand, connection to multiple MSSQL Servers works properly:

![obraz](https://user-images.githubusercontent.com/55132446/109129019-6de8e280-7750-11eb-9058-77ce9727404a.png)

**Reply by thetutlage** [reply]
Tried it myself and seems to be working fine for me. Need more help from you to debug the issue. Can you please check if the following path exists or not?

```
./node_modules/knex-dynamic-connection/build/src/dialects/oracledb.js
```

**Reply by mikelogaciuk** [reply]
The file exists and it's not empty:
```
./node_modules/knex-dynamic-connection/build/src/dialects/oracledb.js
```
![obraz](https://user-images.githubusercontent.com/55132446/109154291-60425580-776e-11eb-8aed-e01787a63c6d.png)

Which version of TS are you using? Mine is 4.1.5.

**Reply by mikelogaciuk** [reply]
I found an error, just wasn't looking properly:

![obraz](https://user-images.githubusercontent.com/55132446/109156210-dba50680-7770-11eb-8074-106c4e5f6329.png)

I don't know why it installed as `orcaledb`, instead of `oracledb`, while node searches for `./src/dialects/oracledb`.

Now it works:

![obraz](https://user-images.githubusercontent.com/55132446/109156387-11e28600-7771-11eb-955e-360ba38f9a63.png)

**Reply by RomainLanz** [reply]
Thanks for your troubleshoot!

It should be fixed soon. (https://github.com/thetutlage/knex-dynamic-connection/pull/6)

**Reply by thetutlage** [reply]
Yup. Fixed and release as `knex-dynamic-connection@1.0.6`. I believe removing the `package-lock.json` file and `node_modules` and then running `npm install` should pull the latest version

**Reply by mikelogaciuk** [reply]
Yes, now the typo doesn't occur.

Thanks a lot.
